contrib/bsp/src/mill/contrib/BSP.scala

```diff
@@ -0,0 +1,132 @@
+package mill.contrib
+
+import java.io.{File, PrintWriter}
+
+import play.api.libs.json._
+import java.nio.file.FileAlreadyExistsException
+import java.util.concurrent.Executors
+
+import upickle.default._
+import ch.epfl.scala.bsp4j._
+import mill._
+import mill.define.{Command, Discover, ExternalModule}
+import mill.eval.Evaluator
+import org.eclipse.lsp4j.jsonrpc.Launcher
+
+import scala.collection.JavaConverters._
+import scala.concurrent.CancellationException
+
+
+object BSP extends ExternalModule {
+
+  implicit def millScoptEvaluatorReads[T] = new mill.main.EvaluatorScopt[T]()
+
+  lazy val millDiscover: Discover[BSP.this.type] = Discover[this.type]
+  val version = "1.0.0"
+  val bspVersion = "2.0.0"
```

Comment:
What's the difference between `version` and `bspVersion`? Do we need both?

Reply:
Here `version` refers to the version of the server (contrib.bsp), whereas `bspVersion` means the version of the protocol used.